packages/tfjs-node-helpers/src/testing/metrics/roc-auc.ts

```diff
@@ -0,0 +1,40 @@
+import { binarize } from '../../utils/binarize';
+import { Metric } from '../metric';
+import { MetricCalculator, MetricCalculatorData } from '../metric-calculator';
+import { FPRMetricCalculator } from './fpr';
+import { RecallMetricCalculator } from './recall';
+
+export class ROCAUCMetricCalculator extends MetricCalculator {
+  constructor(private thresholdNumber: number = 21) {
+    super();
+  }
+
+  public calculate({ trueValues, probabilities }: MetricCalculatorData): Metric {
+    let area = 0;
+    let threshold = 0;
+    const step = 1 / (this.thresholdNumber - 1);
+    const tprs: Array<number> = [];
+    const fprs: Array<number> = [];
+    const fprCalculator = new FPRMetricCalculator();
+    const tprCalculator = new RecallMetricCalculator();
+
+    for (let i = 0; i < this.thresholdNumber; i++) {
+      const predictedValues = binarize(probabilities, threshold);
+
+      fprs.push(fprCalculator.calculate({ trueValues, probabilities, predictedValues }).value);
+      tprs.push(tprCalculator.calculate({ trueValues, probabilities, predictedValues }).value);
+
+      if (i > 0) {
+        area += ((tprs[i] + tprs[i - 1]) * (fprs[i - 1] - fprs[i])) / 2;
+      }
+
+      threshold += step;
+      predictedValues.dispose();
```

Comment:
@VladimirGlinskii It looks like we could dispose `predictedValues` right after pushing the calculated `tpr`.